integration-tests/matching.ts

```diff
@@ -35,6 +36,195 @@ void test('Pupil Request Match', async () => {
+                        id
+                        dissolved
+                        student { firstname lastname }
+                    }
+                }
+            }
+        }
+    `);
+
+    assert.strictEqual(p1.openMatchRequestCount, 0);
+    assert.strictEqual(p1.matches.length, 1);
+    assert.strictEqual(p1.matches[0].dissolved, false);
+    assert.strictEqual(p1.matches[0].student.firstname, student.firstname);
+    assert.strictEqual(p1.matches[0].student.lastname, student.lastname);
+
+    const { me: { student: s1 }} = await studentClient.request(`
+        query StudentWithMatch {
+            me {
+                student {
+                    openMatchRequestCount
+                    matches {
+                        id
+                        dissolved
+                        pupil { firstname lastname }
+                    }
+                }
+            }
+        }
+    `);
+
+    assert.strictEqual(s1.openMatchRequestCount, 0);
+    assert.strictEqual(s1.matches.length, 1);
+    assert.strictEqual(s1.matches[0].dissolved, false);
+    assert.strictEqual(s1.matches[0].pupil.firstname, pupil.firstname);
+    assert.strictEqual(s1.matches[0].pupil.lastname, pupil.lastname);
+
+    return { id: s1.matches[0].id, pupil, student, pupilClient, studentClient };
+});
+
+void test('Create Chat for Match', async () => {
+    const { id, pupil, student, pupilClient, studentClient } = await match1;
+
+    // Creating a Chat ensures all users have accounts:
+
+    // The pupil does not have one, it gets created:
+    expectFetch({
+        method: "GET",
+        url: `https://api.talkjs.com/v1/mocked-talkjs-appid/users/pupil_${pupil.pupil.id}`,
+        responseStatus: 404
+    });
+
+    // The student has one:
+    expectFetch({
+            method: "GET",
+            url: `https://api.talkjs.com/v1/mocked-talkjs-appid/users/student_${student.student.id}`,
+            responseStatus: 200,
+            response: {} // TODO: Mock properly
+    });
+
+    expectFetch({
+        url: `https://api.talkjs.com/v1/mocked-talkjs-appid/users/pupil_${pupil.pupil.id}`,
+        "method": "PUT",
+        "body": JSON.stringify({"name": `${pupil.firstname} ${pupil.lastname}`, "email": [pupil.email.toLowerCase()], "role": "pupil"}),
+        "responseStatus": 200
+    });
+
+    expectFetch({
+        method: "GET",
+        url: `https://api.talkjs.com/v1/mocked-talkjs-appid/users/pupil_${pupil.pupil.id}`,
+        responseStatus: 200,
+        response: {} // TODO: Mock properly
+    });
+
+
+    // Then the conversion is created:
+
+    expectFetch({
+        url: "https://api.talkjs.com/v1/mocked-talkjs-appid/conversations/*",
+        method: "GET",
+        responseStatus: 404
+      });
+
+        // TODO: Remove duplicate fetch
+      expectFetch({
+        url: "https://api.talkjs.com/v1/mocked-talkjs-appid/conversations/*",
+        method: "GET",
+        responseStatus: 404
+      });
+
+      // TODO: Why PUT twice?
```

Comment:
The API design of TalkJS is a bit weird, they use PUT for creating (instead of POST) and updating conversations